src/test/rustdoc-ui/intra-links-warning.rs

```diff
@@ -7,9 +7,9 @@
 //~^ WARNING `Uniooon::X`
 //~| WARNING `Qux::Z`
        //!
-      //! , [Uniooon::X] and [Qux::Z].
-//~^ WARNING `Uniooon::X`
-//~| WARNING `Qux::Z`
+      //! , [Uniooon::Y] and [Qux::W].
+//~^ WARNING `Uniooon::Y`
+//~| WARNING `Qux::W`
```

Comment:
@Manishearth do you know what this is meant to test? Is it that you get a warning on different items?

If it's just testing the warning is emitted at all I'd rather use the cases in `intra-link-errors.rs` I think.